service-tests/json.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'json', title: 'User Defined JSON Source Data' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('/Package Name/$.name/https://github.com/badges/shields/raw/master/package.json.json')
+  .networkOff()
+  .expectJSON({ name: 'Package Name', value: 'inaccessible' });
+
+t.create('JSON from url')
+  .get('/Package Name/$.name/https://github.com/badges/shields/raw/master/package.json.json')
```

Comment:
These tests could use a hard-coded JS object / JSON string and `.intercept()`. We don't really gain anything from hitting the network. Same with the XML tests. An XML string in the test file, with `intercept()`, should suffice.

Reply:
Good idea